### ISSUE TITLE
Low accuracy of LeNet

### ISSUE THREAD

**Issue by llCurious** [ucbrise/piranha#11]
Hey. I try to run the experiments for LeNet model. However i noticed that the convergence of LeNet is weird. The test accuracy of LeNet remains low (far from 98%). I only changed the `localhost_config.json` to run locally with the following configurations:

1. 3PC computation based on Falcon
2. network: `files/models/lenet.json`
3. batch size: 128
4. epoch: 5
5. lr: 2^-4
6. precision: 16

The output is as follows:
![image](https://user-images.githubusercontent.com/21785695/230373192-33fd6689-e9fa-4d32-8363-9992c3db185d.png)

**Comment by llCurious** [comment]
Hey, @jlwatson. Sry to bother you. But do you have any ideas?

**Comment by jlwatson** [comment]
Hey! I can't immediately run some tests, but as a starting point, could you try running it with `models/lenet-norelu.json`?

That's the actual LeNet architecture we evaluated with, it just doesn't have the last relu layer, which doesn't interact well with our loss function.

**Comment by llCurious** [comment]
Thanks to you quick reply!!
I just run the model of lenet-norelu. The accuracy results seems to work fine. BTW, do you mean for other networks like secureml, minionn and so on, I should remove the last relu layer?
![image](https://user-images.githubusercontent.com/21785695/230584859-4a692a36-00b1-46ab-8b7c-22b7b24b8c7d.png)

**Comment by llCurious** [comment]
I just check the figures on the paper. The presented figure is only about 97% for training 10 epochs. The above figures that I obtain is for 5 epochs. Is there something wrong with this part?
![image](https://user-images.githubusercontent.com/21785695/230585340-9c2b1514-730d-4432-a4a6-ca75932d5a23.png)

**Comment by jlwatson** [comment]
Hey just to follow up, you should use the versions without the last ReLU
layer -- it plays nicer with our loss function. And LeNet is pretty easy to
train, so I'm not very surprised to see high accuracy after only 5 rounds.
Looks like things are working properly.

**Comment by allenlin316** [comment]
I also tried the same training using `local_running.sh` with MNIST dataset with `files/models/lenet-norelu.json` model. But somehow I only got the result as below image. 
I wonder what did I do wrong? Thanks in advance!
![image](https://github.com/user-attachments/assets/a4179308-179e-4d3d-9757-6b11c6774762)

**Comment by allenlin316** [comment]
> I also tried the same training using `local_running.sh` with MNIST dataset with `files/models/lenet-norelu.json` model. But somehow I only got the result as below image. I wonder what did I do wrong? Thanks in advance! ![image](https://private-user-images.githubusercontent.com/79969875/380616456-a4179308-179e-4d3d-9757-6b11c6774762.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzAxMDg5OTIsIm5iZiI6MTczMDEwODY5MiwicGF0aCI6Ii83OTk2OTg3NS8zODA2MTY0NTYtYTQxNzkzMDgtMTc5ZS00ZDNkLTk3NTctNmIxMWM2Nzc0NzYyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDI4VDA5NDQ1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzMWJhZDBmZWRmYWMwZmFmOWM2ZmNkY2Q0ZmQ3ZGFlZTIxOTA4OWQ5MDQwNjA0YzNlNDM5NjdlMjgxODVkMTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.CUVGhnOTCfovTVbgNvZCz3G7rr55JYLIXA-DWtgftJI)

I just solved it by remaking the Makefile with 23 bits because of paper mentioned LeNet secure training of neural network 
command remake is as:  `make -j8 PIRANHA_FLAGS="-DFLOAT_PRECISION=23 -DTWOPC"`
